adal/src/main/java/com/microsoft/aad/adal/WebRequestHandler.java

```diff
@@ -83,10 +83,10 @@ public HttpWebResponse sendPost(URL url, Map<String, String> headers, byte[] con
         return request.send();
     }
 
-    private Map<String, String> updateHeaders(Map<String, String> headers) {
+    private Map<String, String> updateHeaders(final Map<String, String> headers) {
 
         if (headers == null) {
-            headers = new HashMap<>();
+            throw new IllegalArgumentException("headers");
         }
```

Comment:
just delete this check.

---
In reply to: [71032456](https://github.com/AzureAD/azure-activedirectory-library-for-android/pull/684#discussion_r71032456) [](ancestors = 71032456)